### PR TITLE
Add Z3 SanityCheck, revert early sanity change

### PR DESCRIPTION
**Pull request by Ludy87**

### CODE REVIEW

**Comment by GMagician** [comment]
I think PIO4 change has to wait...I already posted it some time ago and decision was to give time to all guys to update to 4 before update it (env_default is deprecated, not removed)

**Comment by Ludy87** [comment]
Okay, I did not see that.
But I think 14 days are enough time after the release.

https://docs.platformio.org/en/latest/history.html#id2

File: Marlin/src/inc/SanityCheck.h [hunk 1]
```diff
@@ -2060,24 +2061,22 @@ static_assert(Y_MAX_LENGTH >= Y_BED_SIZE, "Movement bounds (Y_MIN_POS, Y_MAX_POS
...
-      #error "SENSORLESS_HOMING requires X_MAX_ENDSTOP_INVERTING and ENDSTOPPULLUP_XMAX when homing to X_MAX."
-    #elif Y_SENSORLESS && Y_HOME_DIR < 0 && (Y_MIN_ENDSTOP_INVERTING != Y_ENDSTOP_INVERTING || DISABLED(ENDSTOPPULLUP_YMIN))
-      #error "SENSORLESS_HOMING requires Y_MIN_ENDSTOP_INVERTING and ENDSTOPPULLUP_YMIN when homing to Y_MIN."
-    #elif Y_SENSORLESS && Y_HOME_DIR > 0 && (Y_MAX_ENDSTOP_INVERTING != Y_ENDSTOP_INVERTING || DISABLED(ENDSTOPPULLUP_YMAX))
-      #error "SENSORLESS_HOMING requires Y_MAX_ENDSTOP_INVERTING and ENDSTOPPULLUP_YMAX when homing to Y_MAX."
-    #elif Z_SENSORLESS && Z_HOME_DIR < 0 && (Z_MIN_ENDSTOP_INVERTING != Z_ENDSTOP_INVERTING || DISABLED(ENDSTOPPULLUP_ZMIN))
-      #error "SENSORLESS_HOMING requires Z_MIN_ENDSTOP_INVERTING and ENDSTOPPULLUP_ZMIN when homing to Z_MIN."
-    #elif Z_SENSORLESS && Z_HOME_DIR > 0 && (Z_MAX_ENDSTOP_INVERTING != Z_ENDSTOP_INVERTING || DISABLED(ENDSTOPPULLUP_ZMAX))
-      #error "SENSORLESS_HOMING requires Z_MAX_ENDSTOP_INVERTING and ENDSTOPPULLUP_ZMAX when homing to Z_MAX."
-    #endif
```
Comment (thinkyhead) [inline Marlin/src/inc/SanityCheck.h:2080]: Oops, I guess this isn't pertinent yet. The change comes from 14044.

**Comment by thinkyhead** [comment]
> But I think 14 days are enough time after the release.

Marlin is not yet wholly dependent on PIO 4.0.x. and this little change completely breaks 3.x compatibility. So it is better to wait until `env_default` is not just _deprecated_ but is on the verge of being completely removed.

**Pull request merged**